### PR TITLE
CB-56 Use the status of patients in searching (dead or alive)

### PR DESCRIPTION
**Pull request by andela-aolaniran**
# JIRA TICKET NAME:
CB-56 [Use the status of patients in searching (dead or alive)](https://issues.openmrs.org/browse/CB-56)
## SUMMARY:
In the search by demographics form, users can now search for patients based on their living status (alive/dead).

### CODE REVIEW

File: app/js/components/tabs/tabcomponents/patientComponent.jsx [hunk 1]
```diff
@@ -55,7 +57,22 @@ class PatientComponent extends Component {
...
             delete searchParameters.ageRangeOnDate;
         }
-
+        // for dead people, diedDuring period -> startDate == now && endDate == the begining of the world
+        // for living people, diedDuring period -> startDate == now && endDate > startDate
+        const today = new Date();
+        const dayFormat = `${today.getFullYear()}-${today.getMonth() + 1}-${today.getDate()}`;
+        if (this.state.livingStatus === 'alive') {
+            searchParameters.diedDuringPeriod = [
+                { name: 'endDate', dataType: 'date', value: dayFormat, livingStatus: 'alive' }
```
Comment (femioladeji) [inline app/js/components/tabs/tabcomponents/patientComponent.jsx]: What if you extract the common things in line 66 and 73 into a single statement e.g 
```
searchParameters.diedDuringPeriod = [
    { name: 'endDate', dataType: 'date', value: dayFormat }
}
```
then you can now set the livingStatus property in the conditions e.g `searchParameters.diedDuringPeriod[0].livingStatus = 'alive'`

File: app/js/helpers/jsonHelper.js [hunk 2]
```diff
@@ -108,7 +118,10 @@ export class JSONHelper {
       let counter = 0;
       for(let eachKey in searchParameters) {
         if(counter === 0) {
-          label += ' with';
+          if (eachKey !== 'diedDuringPeriod')
+            label += ' with';
+          else
+            label += ' that are'
```
Comment (femioladeji) [inline app/js/helpers/jsonHelper.js]: Don't forget the semicolon

File: app/js/helpers/jsonHelper.js [hunk 3]
```diff
@@ -56,7 +60,13 @@ export class JSONHelper {
     let compositionTitle = '';
     const totalNumber = filterColumns.length;
     for (let index = 1; index <= totalNumber; index++) {
-      compositionTitle += `${index}`;
+      if (filterColumns[index - 1].livingStatus === 'alive') {
+        compositionTitle += `NOT ${index}`;
+        // delete helper living status property
+        delete filterColumns[index - 1].livingStatus;
```
Comment (femioladeji) [inline app/js/helpers/jsonHelper.js:66]: Smart move

**Comment by femioladeji** [comment]
LGTM

**Comment by ghost** [comment]
LGTM

**Pull request closed without merging**